src/imp/exec/exec.c

```diff
@@ -381,8 +407,15 @@ int imp_exec_unprivileged (struct imp_state *imp, struct kv *kv)
         imp_die (1, "exec: user %s not in allowed-users list",
                     exec->imp_pwd->pw_name);
 
-    /* Read input from stdin, cmdline: */
-    imp_exec_init_stream (exec, stdin);
+    if ((helper = getenv ("FLUX_IMP_EXEC_HELPER"))
+        && strlen (helper) > 0) {
+        /* Read input from helper command */
+        imp_exec_init_helper (exec, helper);
+    }
+    else {
+        /* Read input from stdin, cmdline: */
+        imp_exec_init_stream (exec, stdin);
+    }
```

Comment:
Might be worth noting why an empty string value for FLUX_IMP_EXEC_HELPER is not treated as an error?

Reply:
Actually that's a good question. Maybe it should just be an error? For some reason I was thinking it was idiomatic to have an empty environment variable be treated as unset, as it makes things like this possible `VARIABLE= command` to "unset" a possibly set `VARIABLE` for the execution of `command`. Maybe I'll just say that in the comment.

Reply:
Upon deep reflection, probably it is most clear and safest to return an error if `FLUX_IMP_EXEC_HELPER` is empty. This wasn't used in testing so it is easy to make this change.